packages/client-utils/src/sequence-manager.ts

```diff
@@ -0,0 +1,59 @@
+import type { BaseAccount } from './codegen/cosmos/auth/v1beta1/auth.js';
+
+/**
+ * @alpha
+ */
+export type TxSequencer = {
+  getAccountNumber: () => bigint;
+  getSequenceNumber: () => bigint;
+  resync: () => Promise<void>;
+};
+
+/**
+ * Manage sequence numbers for submitting transactions from a single address.
+ *
+ * @alpha
+ */
+export const makeTxSequencer = async (
+  fetchAccount: () => Promise<BaseAccount>,
+  { log = () => {} }: { log?: (...args: unknown[]) => void },
+): Promise<TxSequencer> => {
+  const initial = await fetchAccount();
+  const accountNumber: bigint = initial.accountNumber;
+  let sequenceNumber: bigint = initial.sequence;
+  log(
+    `Initialized accountNumber ${accountNumber} sequence number to ${sequenceNumber}`,
+  );
+
+  /**
+   * Get the account number.
+   */
+  const getAccountNumber = () => accountNumber;
+
+  /**
+   * Get the next sequence number and increment the internal counter.
+   */
+  const getSequenceNumber = () => {
+    const curr = sequenceNumber;
+    sequenceNumber += 1n;
+    return curr;
+  };
```

Comment:
On second thought, I don't think a function with side effects that modify internal state should be named "get". It's really analogous to an Iterator `next` method.
```suggestion
  /**
   * Return the sequence number expected for the next transaction and increment
   * the internal counter for subsequent invocations.
   */
  const nextSequenceNumber = () => {
    const curr = sequenceNumber;
    sequenceNumber += 1n;
    return curr;
  };
```